### ISSUE TITLE
Stretched volumes after RandomRotate for uneven dimensionalities (3D)

### ISSUE THREAD

**Issue by eikekutz** [alexandrainst/torch-trandsforms#115]
## 🐛 Bug Report

When using the trandsforms function RandomRotate on volumes with uneven dimensionalities ( in this example 64x64x128) the output doesn't look rotated correctly and furthermore looks transformed in an unwanted way.
When the input in padded with zero voxels to create an input shape of 128x128x128 the transformation looks to be correct. 
 

## 🔬 How To Reproduce

1. See coding example. For p=0.0 no transformation should apply and p=1.0 a transformation with should apply.
2. the example object is set to touch the boundaries of the volume in two dimensions
3. check output of the transformation

### Version 1. p =0.0 no transformation
```python

import numpy as np
import torch
from torch_trandsforms import RandomRotate, Compose
from trimesh.voxel.base import VoxelGrid
from trimesh import Scene


transform = Compose([
        RandomRotate([180,180,180], sample_mode="nearest", p=0.0, nd=3, keys=["input"]),
        ])

dummy_cube = torch.zeros((1, 64, 64, 128))
dummy_cube[:, :,:,32:96] = 1

transformed_cube = transform(input=dummy_cube)['input'].numpy()

vol_colors = np.zeros((*transformed_cube[0].shape, 4))
vol_colors[:,:,:,:] = [0, 0, 1, 0.1]
displayable_volume = VoxelGrid(transformed_cube[0]).as_boxes(colors=vol_colors)

scene = Scene()
scene.add_geometry(displayable_volume)
scene.show()
```

### Version 2. p =1.0 with transformation and the bug
```python
import numpy as np
import torch
from torch_trandsforms import RandomRotate, Compose
from trimesh.voxel.base import VoxelGrid
from trimesh import Scene


transform = Compose([
        RandomRotate([180,180,180], sample_mode="nearest", p=1.0, nd=3, keys=["input"]),
        ])

dummy_cube = torch.zeros((1, 64, 64, 128))
dummy_cube[:, :,:,32:96] = 1

transformed_cube = transform(input=dummy_cube)['input'].numpy()

vol_colors = np.zeros((*transformed_cube[0].shape, 4))
vol_colors[:,:,:,:] = [0, 0, 1, 0.1]
displayable_volume = VoxelGrid(transformed_cube[0]).as_boxes(colors=vol_colors)

scene = Scene()
scene.add_geometry(displayable_volume)
scene.show()
```
### Version 3. p =1.0 with transformation and padding to work around the bug
```python
import numpy as np
import torch
from torch_trandsforms import RandomRotate, Compose
from trimesh.voxel.base import VoxelGrid
from trimesh import Scene


transform = Compose([
        RandomRotate([180,180,180], sample_mode="nearest", p=1.0, nd=3, keys=["input"]),
        ])

dummy_cube = torch.zeros((1, 64, 64, 128))
dummy_cube[:, :,:,32:96] = 1
# apply padding to create equal dimensions
dummy_cube = torch.nn.functional.pad(dummy_cube, (0,0,32, 32, 32, 32), mode='constant', value=0)

transformed_cube = transform(input=dummy_cube)['input'].numpy()

vol_colors = np.zeros((*transformed_cube[0].shape, 4))
vol_colors[:,:,:,:] = [0, 0, 1, 0.1]
displayable_volume = VoxelGrid(transformed_cube[0]).as_boxes(colors=vol_colors)

scene = Scene()
scene.add_geometry(displayable_volume)
scene.show()
```
### Environment

* OS: [e.g. Linux & macOS]
* Python version 3.11.11
* torch-trandsforms  0.4.0
* numpy              2.2.2
* torch              2.5.1
* trimesh            4.6.1

### Screenshots

<img width="467" alt="Image" src="https://github.com/user-attachments/assets/99ffda2f-91c6-401d-9fb5-67b496f86175" />
<img width="617" alt="Image" src="https://github.com/user-attachments/assets/8e056ee7-dccb-49b2-a9b3-9352e8a5b9a5" />

## 📈 Expected behavior

Correct rotation potentially within the limits of the volume or user defined padding when relevant.
